monai/data/dataset.py

```diff
@@ -289,3 +290,99 @@ def __getitem__(self, index):
             # no cache for this data, execute all the transforms directly
             data = super(CacheDataset, self).__getitem__(index)
         return data
+
+
+class ArrayDataset(Dataset):
```

Comment:
this class inherits Dataset but is not using its members, perhaps use `self.data` instead of `self.img_files`?